### PR TITLE
polish(clickhouse rest api):  error message, log and compression

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1.   better error message for NDJson,  show types.
2.  log HTTP handler requests (without the body, enough most time)
3.  clickhouse rest api support compression.

## Changelog

- Not for changelog (changelog entry is not required)

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/4622

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Review by BohuTANG — APPROVED** [review]
lgtm

File: common/streams/src/sources/source_ndjson.rs [hunk 1]
```diff
@@ -123,11 +123,19 @@ where R: AsyncBufRead + Unpin + Send
...
+            for ((name, type_name), deser) in fields.iter().zip(packs.iter_mut()) {
                 let value = &json[name];
-                deser.de_json(value)?;
+                deser.de_json(value).map_err(|e| {
+                    ErrorCode::BadBytes(format!(
+                        "error at row {} column {}: type={}, value={:?}: {}",
+                        rows,
+                        name,
+                        type_name,
+                        value,
```
Comment (BohuTANG) [inline common/streams/src/sources/source_ndjson.rs:135]: If the value is larger, we need to truncate it to a MAX_LENGTH(1024)

Comment (youngsofun) [reply common/streams/src/sources/source_ndjson.rs:135]: 👌

Comment (youngsofun) [reply common/streams/src/sources/source_ndjson.rs:135]: fixed in https://github.com/datafuselabs/databend/pull/4640

**Pull request merged**